src/ns_tracker/core.clj

```diff
@@ -34,32 +39,34 @@
   (with-open [rdr (PushbackReader. (io/reader file))]
     (func rdr)))
 
-(defn- newer-namespace-decls [then now]
+(defn- newer-namespace-decls [then now dependency-graph]
```

Comment:
Instead of calling it `dependency-graph`, what about `extra-deps`?

Also, does this need to be an argument or can it be a global? My recollection is that the Clojure namespace dependencies are handled by tools.namespace as a global var.

Reply:
Does this dependency graph include only the additional resource dependencies, or does it include the dependencies handled by tools.namespace as well?

Reply:
Here is an example of what the dependency graph looks like when executing ns-tracker's tests:

```clj
{:dependencies {example.x #{example.internal.y example.internal.z},
                example.db #{#<File tmp/sql/queries.sql>},
                example.core #{example.util},
                example.d #{example.c},
                example.b #{example.a},
                example.c #{example.b}},
 :dependents {example.internal.z #{example.x},
              example.internal.y #{example.x},
              #<File tmp/sql/queries.sql> #{example.db},
              example.util #{example.core},
              example.c #{example.d},
              example.a #{example.b},
              example.b #{example.c}}}
```

Reply:
ns-tracker uses only a couple parsing functions from clojure.tools.namespace. It doesn't rely on clojure.tools.namespace to track any dependencies.

Reply:
Ah, I see. Sorry, I got my wires crossed and confused this with another project that tracks dependencies. Please ignore this whole comment chain :)